### PR TITLE
pytest migration of unit.common.metadata.test_metadata_manager_factory

### PR DESCRIPTION
**Pull request by bjlittle**
### Description

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

---
Add any of the below labels to trigger actions on this PR:

- https://github.com/SciTools/iris/labels/benchmark_this

### CODE REVIEW

File: lib/iris/tests/unit/common/metadata/test_metadata_manager_factory.py [hunk 1]
```diff
@@ -156,40 +156,38 @@ def setUp(self):
...
         self.bases = BASES
 
     def test(self):
         for base in self.bases:
             fields = base._fields
             metadata = metadata_manager_factory(base)
-            self.assertEqual(metadata.fields, fields)
+            assert metadata.fields == fields
             for field in fields:
                 hasattr(metadata, field)
```
Comment (pp-mo) [inline lib/iris/tests/unit/common/metadata/test_metadata_manager_factory.py:180]: Not a pytest issue, but I can't help being puzzled by this "hasattr" bit.
Is it a mistake, or does it merely test for "does not crash" ?

**Review by pp-mo — APPROVED** [review]
One silly question, otherwise all good :+1:

File: lib/iris/tests/unit/common/metadata/test_metadata_manager_factory.py (hunk 1 above)
Comment (bjlittle) [reply lib/iris/tests/unit/common/metadata/test_metadata_manager_factory.py:180]: I guess it's a pattern that I used to ensure that the fields are genuinely accessible.

We could add an `assert` here

Comment (bjlittle) [reply lib/iris/tests/unit/common/metadata/test_metadata_manager_factory.py:180]: See lines +62 and +64 ... yup, you've spotted a genuine omission 👍

**Pull request merged**